react/features/transcribing/middleware.js

```diff
@@ -0,0 +1,99 @@
+// @flow
+
+import { MiddlewareRegistry } from '../base/redux';
+
+import {
+    _TRANSCRIBER_LEFT,
+    DIAL_TRANSCRIBER,
+    STOP_TRANSCRIBING
+} from './actionTypes';
+import {
+    potentialTranscriberJoined,
+    transcriberJoined,
+    transcriberLeft,
+    dialError,
+    showPendingTranscribingNotification,
+    hidePendingTranscribingNotification,
+    showStoppedTranscribingNotification,
+    showTranscribingError
+} from './actions';
+import {
+    HIDDEN_PARTICIPANT_JOINED,
+    HIDDEN_PARTICIPANT_LEFT,
+    PARTICIPANT_UPDATED
+} from './../base/participants';
+
+declare var APP: Object;
+
+const TRANSCRIBER_DIAL_COMMAND = 'jitsi_meet_transcribe';
+const TRANSCRIBER_DISPLAY_NAME = 'Transcriber';
+
+/**
+ * Implements the middleware of the feature transcribing.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+// eslint-disable-next-line no-unused-vars
+MiddlewareRegistry.register(store => next => action => {
+    const {
+        isDialing,
+        isTranscribing,
+        transcriberJID,
+        potentialTranscriberJIDs
+    } = store.getState()['features/transcribing'];
+
+    switch (action.type) {
+    case DIAL_TRANSCRIBER:
+        if (!(isDialing || isTranscribing)) {
+            store.dispatch(showPendingTranscribingNotification());
+
+            APP.conference._room.dial(TRANSCRIBER_DIAL_COMMAND).then(
+                () => {
+                    // do nothing
```

Comment:
Maybe using a .catch instead of .then is more expressive of what you want to do.